File: src/main/java/sh/ball/shapes/QuadraticBezierCurve.java
package sh.ball.shapes;

public class QuadraticBezierCurve extends CubicBezierCurve {

  public QuadraticBezierCurve(Vector2 p0, Vector2 p1, Vector2 p2) {
    super(p0, p0.add(p1.sub(p0).scale(2.0/3.0)), p2.add(p1.sub(p2).scale(2.0/3.0)), p2);
  }
}

File: Source/shape/QuadraticBezierCurve.cpp
#include "QuadraticBezierCurve.h"

QuadraticBezierCurve::QuadraticBezierCurve(double x1, double y1, double x2, double y2, double x3, double y3) : CubicBezierCurve(x1, y1, x1 + (x2 - x1) * (2.0 / 3.0), y1 + (y2 - y1) * (2.0 / 3.0), x3 + (x2 - x3) * (2.0 / 3.0), y3 + (y2 - y3) * (2.0 / 3.0), x3, y3) {}
